main.cpp

```diff
@@ -92,6 +92,107 @@ good to go!
  */
 
 #include <iostream>
+
+
+
+struct FloatType
+{
+    float add(float, float);
+    float subtract(float, float);
+    float multiply(float, float);
+    float divide(float, float);
+};
+
+float FloatType::add(float lhs, float rhs)
+{
+    return lhs+rhs;
+}
+
+float FloatType::subtract(float lhs, float rhs)
+{
+    return lhs-rhs;
+}
+
+float FloatType::multiply(float lhs, float rhs)
+{
+    return lhs*rhs;
+}
+
+float FloatType::divide(float lhs, float rhs)
+{
+    if(rhs==0.0f)
+    {
+        std::cout  << "\nwarning, floating point division by zero returns 'inf' !\n";
+    }
+    return lhs/rhs;
+}
+
+struct DoubleType
+{
+    double add(double, double);
+    double subtract(double, double);
+    double multiply(double, double);
+    double divide(double, double);
+};
+
+double DoubleType::add(double lhs, double rhs)
+{
+    return lhs+rhs;
+}
+
+double DoubleType::subtract(double lhs, double rhs)
+{
+    return lhs-rhs;
+}
+
+double DoubleType::multiply(double lhs, double rhs)
+{
+    return lhs*rhs;
+}
+
+double DoubleType::divide(double lhs, double rhs)
+{
+    if(rhs==0.0)
+    {
+        std::cout  << "\nwarning, floating point division by zero returns 'inf' !\n";
+    }
+    return lhs/rhs;
+}
```

Comment:
Add space around your math operators so your code is more readable
- [ ] revise